zipline/data/history_loader.py

```diff
@@ -573,8 +573,9 @@ def _frequency(self):
     @lazyval
     def _calendar(self):
         mm = self.trading_calendar.all_minutes
-        return mm[mm.slice_indexer(start=self._reader.first_trading_day,
-                                   end=self._reader.last_available_dt)]
+        start = mm.searchsorted(self._reader.first_trading_day)
```

Comment:
I agree that we should. Will follow up with a refactoring of this and the other use of the same pattern.